File: algorithms/02_c/data_structures/stack.c
#include <stdio.h>
#define MAX 1000
typedef struct { int data[MAX]; int top; } Stack;
void init(Stack *s){ s->top=-1; }
void push(Stack *s, int v){ s->data[++s->top]=v; }
int  pop(Stack *s){ return s->data[s->top--]; }
int  peek(Stack *s){ return s->data[s->top]; }
int  empty(Stack *s){ return s->top==-1; }

File: algorithms/11_golang/data_structures/stack.go
package data_structures
type Stack[T any] struct { data []T }
func (s *Stack[T]) Push(v T) { s.data=append(s.data,v) }
func (s *Stack[T]) Pop() T { n:=len(s.data)-1; v:=s.data[n]; s.data=s.data[:n]; return v }
func (s *Stack[T]) Peek() T { return s.data[len(s.data)-1] }
func (s *Stack[T]) Empty() bool { return len(s.data)==0 }